nifi-toolkit/nifi-toolkit-encrypt-config/src/main/groovy/org/apache/nifi/toolkit/encryptconfig/EncryptConfigMain.groovy

```diff
@@ -0,0 +1,145 @@
+                "Unprotected files can be input to this tool to be protected by a key in a manner that is understood by NiFi. " +
+                "Protected files, along with a key, can be input to this tool to be unprotected, for troubleshooting or automation purposes.\n\n"
+
+        def options = new Options()
+        options.addOption("h", "help", false, "Show usage information (this message)")
+        options.addOption(null, NIFI_REGISTRY_OPT, false, "Specifies to target NiFi Registry. When this flag is not included, NiFi is the target.")
+
+        HelpFormatter helpFormatter = new HelpFormatter()
+        helpFormatter.setWidth(160)
+        helpFormatter.setOptionComparator(null)
+        helpFormatter.printHelp("${EncryptConfigMain.class.getCanonicalName()} [-h] [options]", header, options, "\n")
+        System.out.println()
+
+        helpFormatter.setSyntaxPrefix("") // disable "usage: " prefix for the following outputs
+
+        Options nifiModeOptions = ConfigEncryptionTool.getCliOptions()
+        helpFormatter.printHelp(
+                "When targeting NiFi:",
+                nifiModeOptions,
+                false)
+        System.out.println()
+
+        Options nifiRegistryModeOptions = NiFiRegistryMode.getCliOptions()
+        nifiRegistryModeOptions.addOption(null, DECRYPT_OPT, false, "Can be used with -r to decrypt a previously encrypted NiFi Registry Properties file. Decrypted content is printed to STDOUT.")
+        helpFormatter.printHelp(
+                "When targeting NiFi Registry using the ${NIFI_REGISTRY_FLAG} flag:",
+                nifiRegistryModeOptions,
+                false)
+        System.out.println()
+
+//        String footer = """
+//            |
+//            |Encrypt a NiFi Registry properties using a password:
+//            |    encrypt-config -p <password> -b /path/to/nifi/conf/bootstrap.conf -r /path/to/nifi/conf/nifi.properties
+//            |
+//            |""".stripMargin()
+        //helpFormatter.printHelp("Examples:", "", new Options(), footer)
+    }
+
+    static void printUsageAndExit(String message = "", int exitStatusCode) {
+        printUsage(message)
+        System.exit(exitStatusCode)
+    }
+
+    static void main(String[] args) {
+        Security.addProvider(new BouncyCastleProvider())
+
+        if (args.length < 1) {
+            printUsageAndExit(EXIT_STATUS_FAILURE)
+        }
+
+        String firstArg = args[0]
+
+        if (["-h", "--help"].contains(firstArg)) {
+            printUsageAndExit(EXIT_STATUS_OTHER)
+        }
+
+        try {
+            List<String> argsList = args
+            ToolMode toolMode = determineModeFromArgs(argsList)
+            if (toolMode) {
+                toolMode.run((String[])argsList.toArray())
+                System.exit(EXIT_STATUS_SUCCESS)
+            } else {
+                printUsageAndExit(EXIT_STATUS_FAILURE)
+            }
+        } catch (Throwable t) {
+            logger.debug("", t)
```

Comment:
Change this to `logger.error()` as well.

Reply:
- [ ] use `logger.error()` in ECM